File: MyApplication/app/src/main/java/com/example/myapplication/Activity2.java
package com.example.myapplication;

import androidx.appcompat.app.AppCompatActivity;

import android.app.ActionBar;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class Activity2 extends MainActivity {
    @Override
    public void getNextButton() {
        Intent i = new Intent(Activity2.this, Activity3.class );
        startActivity(i);
    }

    @Override
    public void getPreviousButton() {
        Intent i = new Intent(Activity2.this, Activity1.class );
        startActivity(i);
    }

    @Override
    public String getAcitivityName() {
        return "Activity 2";
    }
}

File: MyApplication7/app/src/main/java/com/example/myapplication/Activity2.kt
package com.example.myapplication

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_2.*
import kotlinx.android.synthetic.main.activity_main.*

class Activity2 : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_2)

        button2.setOnClickListener {
            startActivity(Intent(this, Activity3::class.java))
        }

        button3.setOnClickListener {
            startActivity(Intent(this, MainActivity::class.java))
        }

        supportActionBar!!.title = "Back"
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)

    }

    override fun onSupportNavigateUp(): Boolean {
        onBackPressed()
        return true
    }
}
